### PR TITLE
net-p2p/tremc: New package

### PR DESCRIPTION
**Pull request by ajakk**
transmission-remote-cli is dead and Python 2 only. This is a fork of it that supports Python 3.

Signed-off-by: John Helmert III <jchelmertt3@gmail.com>
Bug: https://bugs.gentoo.org/713518

### CODE REVIEW

File: net-p2p/tremc/tremc-0.9.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
```
Comment (mgorny) [inline net-p2p/tremc/tremc-0.9.1.ebuild:6]: Why no love for 3.8?

File: net-p2p/tremc/tremc-0.9.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+DESCRIPTION="Ncurses interface for the Transmission BitTorrent client"
+HOMEPAGE="https://github.com/tremc/tremc"
+SRC_URI="https://github.com/tremc/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="geoip"
+
```
Comment (mgorny) [inline net-p2p/tremc/tremc-0.9.1.ebuild:18]: No empty line here, please.

File: net-p2p/tremc/tremc-0.9.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	geoip? ( dev-python/geoip-python[$PYTHON_USEDEP] )
+"
+
+# Specify a nop src_compile so upstream's broken Makefile doesn't get used
+src_compile() {
+	:
+}
+
+src_install() {
+	python_foreach_impl python_doscript tremc
```
Comment (mgorny) [inline net-p2p/tremc/tremc-0.9.1.ebuild:31]: Using python-r1 for this looks like a huge overkill. Use python-single-r1.

File: net-p2p/tremc/tremc-0.9.1.ebuild (hunk 1 above)
Comment (ajakk) [reply net-p2p/tremc/tremc-0.9.1.ebuild:6]: `dev-lang/geoip-python` only has python3_{6,7}, it's my understanding that reverse dependencies are limited to the impls the dependency has.

Comment (mgorny) [reply net-p2p/tremc/tremc-0.9.1.ebuild:6]: Could you test it? That is, if it doesn't turn out to require too many packages.

Comment (ajakk) [reply net-p2p/tremc/tremc-0.9.1.ebuild:6]: Test geoip-python so both can have 3.8 or just tremc?

Comment (ajakk) [reply net-p2p/tremc/tremc-0.9.1.ebuild:6]: In any case, geoip-python tests are successful on python3_{6,7,8}, I can open a new PR with 3.8 added, if you'd like.

Comment (mgorny) [reply net-p2p/tremc/tremc-0.9.1.ebuild:6]: Both. Just add it as another commit to this PR.

File: net-p2p/tremc/tremc-0.9.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+SRC_URI="https://github.com/tremc/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="geoip"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="${PYTHON_DEPS}
+	geoip? ( $(python_gen_cond_dep \
```
Comment (mgorny) [inline net-p2p/tremc/tremc-0.9.1.ebuild:21]: There's no python2 here, so this is redundant.

File: net-p2p/tremc/tremc-0.9.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		'dev-python/geoip-python[${PYTHON_MULTI_USEDEP}]' -3) )
+"
+
+# This fixes a crash when starting that was committed after 0.9.1
+PATCHES=( "${FILESDIR}/${PV}-fix-startup-crash.patch" )
+
+# Specify a nop src_compile so upstream's broken Makefile doesn't get used
```
Comment (mgorny) [inline net-p2p/tremc/tremc-0.9.1.ebuild:28]: NOP is not portable, prefer no-op ;-).

File: net-p2p/tremc/tremc-0.9.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+src_compile() {
+	:
+}
+
+src_install() {
+	python_doscript tremc
+	newbashcomp completion/bash/tremc.sh \
```
Comment (mgorny) [inline net-p2p/tremc/tremc-0.9.1.ebuild:35]: This line doesn't look like it needs wrapping.

**Comment by mgorny** [comment]
Thanks. Overall, this looks good, just a few final touches.

File: net-p2p/tremc/tremc-0.9.1.ebuild (hunk 4 above)
Comment (ajakk) [reply net-p2p/tremc/tremc-0.9.1.ebuild:21]: Sorry? I don't understand what you mean

Comment (mgorny) [reply net-p2p/tremc/tremc-0.9.1.ebuild:21]: The condition is always true, so gen_cond_dep is redundant.

Comment (ajakk) [reply net-p2p/tremc/tremc-0.9.1.ebuild]: Thanks for the clarification

**Comment by ajakk** [comment]
Thanks, sorry for confusion

**Pull request closed without merging**